From f85aad8bf32b442a065b9d1af5b998d4507fe1a4 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sat, 26 Apr 2014 08:55:17 +0300
Subject: [PATCH] Fake cellular modem device ID

In case there isn't one, just fake the ID. This fixes application crashes that doesn't check for null

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 .../java/android/telephony/TelephonyManager.java   |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/telephony/java/android/telephony/TelephonyManager.java b/telephony/java/android/telephony/TelephonyManager.java
index 8f17e72..3a8aa93 100644
--- a/telephony/java/android/telephony/TelephonyManager.java
+++ b/telephony/java/android/telephony/TelephonyManager.java
@@ -249,7 +249,9 @@ public class TelephonyManager {
         } catch (RemoteException ex) {
             return null;
         } catch (NullPointerException ex) {
-            return null;
+            /* Some applications crash if they get null. Fake on */
+            String s = "22222222";
+            return s;
         }
     }
 
-- 
1.7.9.5

